integration/map_test.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"testing"
+
+	"github.com/google/trillian"
+	"github.com/google/trillian/merkle"
+	_ "github.com/google/trillian/merkle/coniks" // Register
+	"github.com/google/trillian/merkle/hashers"
+	_ "github.com/google/trillian/merkle/maphasher" // Register
+	"github.com/google/trillian/testonly"
+	"github.com/google/trillian/testonly/integration"
+)
+
+// createHashKV returns a []*trillian.MapLeaf formed by the mapping of index, value ...
+// createHashKV panics if len(iv) is odd. Duplicate i/v pairs get over written.
+func createMapLeaves(iv ...[]byte) []*trillian.MapLeaf {
+	if len(iv)%2 != 0 {
+		panic(fmt.Sprintf("integration: createMapLeaves got odd number of iv pairs: %v", len(iv)))
+	}
+	r := []*trillian.MapLeaf{}
+	var index []byte
+	for i, b := range iv {
```

Comment:
I don't see how that's better. More LoC, more operations performed, less obvious stepping, so on. I'd nitpick the standard library's PR if I had the chance too. :)

I'll leave it in your hands.